Sources/MapLibreSwiftUI/Models/MapCamera/MapViewCamera.swift

```diff
@@ -115,6 +115,21 @@ public struct MapViewCamera: Hashable {
                              direction: Defaults.direction,
                              lastReasonForChange: .programmatic)
     }
-    
-    // TODO: Create init methods for other camera states once supporting materials are understood (e.g. BoundingBox)
+	
+	/// Positions the camera to show a specific region in the MapView.
+	///
+	/// - Parameters:
+	///   - box: Set the desired bounding box. This is a one time event and the user can manipulate by moving the map.
+	///   - edgePadding: Set the edge insets that should be applied before positioning the map.
+	/// - Returns: The MapViewCamera representing the scenario
+	public static func boundingBox(_ box: MLNCoordinateBounds, edgePadding: UIEdgeInsets = .init(top: 20, left: 20, bottom: 20, right: 20)) -> MapViewCamera {
+		// zoom, pitch & direction are ignored.
+		return MapViewCamera(state: .rect(boundingBox: box, edgePadding: edgePadding),
+							 zoom: 1,
+							 pitch: Defaults.pitch,
+							 direction: Defaults.direction,
+							 lastReasonForChange: .programmatic)
+	}
```

Comment:
I'm a bit unsure about this:

1st how do we pass the edge padding to the camera? This works but I don't know if it's the best solution.

2nd and more importantly: zoom, pitch & direction are ignored for this camera setup. Wouldn't it make more sense to add these parameters as associated values to the respective (CameraState) enum cases? Otherwise adding parameters that are ignored might cause some confusion.

Reply:
### EdgeInsets Padding

I think we're on the right track here, but this presents an interesting trade off we need to discuss:

1. This may suggest we should remove `Hashable` from the MapViewCamera, CameraState, etc. The reason I wanted it was because it allows us to pass state containing a defined `MapViewCamera` through SwiftUI's `NavigationStack`. Basically meaning, you could inject a predefined `MapViewCamera` into a new app screen by passing it through a `NavigationLink`. If we do remove this, it's not the end of the world, it just makes it trickier for a developer to set their initial camera state on presenting a new screen in their app.
2. We could decide it's worth keeping `Hashable` because of the `NavigationLink` value support <https://developer.apple.com/documentation/swiftui/navigationlink/init(_:value:)-kj9v>. In this case, we might want to create our own padding struct (e.g. `ShowcasePadding`) that more easily conforms to `Hashable`, then outputs/converts to a UIEdgeInsets once it's called on the map view.
3. We could keep it as is, in which case my only recommendation is we use SwiftUI's EdgeInsets on this top layer and move the hashable stuff to the right places in the `Sources/Extensions/` folder. Then we'd need to convert `EdgeInsets` to `UIEdgeInsets` down in the coordinator, but that would make it SwiftUI focused on top later/entry point.

This same discussion applies to the bounding box definition type as well `MLNCoordinateBounds`. A lot of the stuff in the underlying MapLibre code is older UIKit based code, so it's not unreasonable to wrap things in a SwiftUI focused top layer.

### Zoom, Pitch & Direction

This is a good question. We had a similar issue deciding how to handle the `.centered` state's coordinates. In the end we decided if the value is only required for certain state(s) we should put it on the `CameraState` enum as a variable. 

I think the boundingBox case suggests we should do the same for `zoom` as well. Since zoom only applies to the `centered` and `tracking...` states. Pitch and direction seem like they should also follow that path _unless_ they actually can be manipulated and maintained when showcasing (e.g. not sure how maplibre handles centering on a bounding box with padding at different compass directions for example, it may always just revert the pitch and direction to default north + straight down?).

Any additional thoughts @ianthetechie?

Reply:
# `EdgeInsets`

> We could decide it's worth keeping Hashable because of the NavigationLink value support https://developer.apple.com/documentation/swiftui/navigationlink/init(_:value:)-kj9v. In this case, we might want to create our own padding struct (e.g. ShowcasePadding) that more easily conforms to Hashable, then outputs/converts to a UIEdgeInsets once it's called on the map view.

I tend to prefer this approach of writing our own padding struct which has a computed property for easy conversion to `UIEdgeInsets` (I don't have any great name ideas myself, but I think `MapInsets` or something makes sense; I understand the `Showcase` term but I'm not sure it's widely used). The reason being that it keeps our public API stable in the event that MapLibre does something else in the future (ex: modernizes) or Apple introduces surprise breakage by adding a protocol conformance.

> This same discussion applies to the bounding box definition type as well MLNCoordinateBounds. A lot of the stuff in the underlying MapLibre code is older UIKit based code, so it's not unreasonable to wrap things in a SwiftUI focused top layer.

Agreed.

# Zoom, Pitch & Direction

> I think the boundingBox case suggests we should do the same for zoom as well. Since zoom only applies to the centered and tracking... states.

Agreed. I'm in favor of moving `zoom` to the `CameraState` as associated values for the relevant variants. In general, the principle we follow is "make illegal state unrepresentable" whenever possible, and you've found a solid shortcoming as it relates to bounding boxes. 

Thanks!